frontend/src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,60 @@
+import { createContext, FC, useCallback, useContext, useState } from 'react'
+
+type AuthContextProps = {
+  user?: any
+  isAuthenticated?: boolean
```

Comment:
mocked for now, will expand on it in the real implementation. The user object will probably be out of this context in the final implementation.

> can't we infer isAuthenticated based on whether user is defined or not? 

Yes we can; in the real implementation we actually store a `is-logged-in` string token in localStorage instead of the user object since we always retrieve the user from our /user endpoint